lib/rubocop/cop/rails/redundant_active_record_all_method.rb

```diff
@@ -0,0 +1,145 @@
+      #
+      # @example
+      #   # bad
+      #   User.all.find(id)
+      #   User.all.order(:created_at)
+      #   users.all.where(id: ids)
+      #   user.articles.all.order(:created_at)
+      #
+      #   # good
+      #   User.find(id)
+      #   User.order(:created_at)
+      #   users.where(id: ids)
+      #   user.articles.order(:created_at)
+      class RedundantActiveRecordAllMethod < Base
+        include ActiveRecordHelper
+        extend AutoCorrector
+
+        MSG = 'Redundant `all` detected.'
+
+        RESTRICT_ON_SEND = [:all].freeze
+
+        # Defined methods in `ActiveRecord::Querying::QUERYING_METHODS` on activerecord 7.0.5.
+        QUERYING_METHODS = %i[
+          and
+          annotate
+          any?
+          average
+          calculate
+          count
+          create_or_find_by
+          create_or_find_by!
+          create_with
+          delete_all
+          delete_by
+          destroy_all
+          destroy_by
+          distinct
+          eager_load
+          except
+          excluding
+          exists?
+          extending
+          extract_associated
+          fifth
+          fifth!
+          find
+          find_by
+          find_by!
+          find_each
+          find_in_batches
+          find_or_create_by
+          find_or_create_by!
+          find_or_initialize_by
+          find_sole_by
+          first
+          first!
+          first_or_create
+          first_or_create!
+          first_or_initialize
+          forty_two
+          forty_two!
+          fourth
+          fourth!
+          from
+          group
+          having
+          ids
+          in_batches
+          in_order_of
+          includes
+          invert_where
+          joins
+          last
+          last!
+          left_joins
+          left_outer_joins
+          limit
+          lock
+          many?
+          maximum
+          merge
+          minimum
+          none
+          none?
+          offset
+          one?
+          only
+          optimizer_hints
+          or
+          order
+          pick
+          pluck
+          preload
+          readonly
+          references
+          reorder
+          reselect
+          rewhere
+          second
+          second!
+          second_to_last
+          second_to_last!
+          select
+          sole
+          strict_loading
+          sum
+          take
+          take!
+          third
+          third!
+          third_to_last
+          third_to_last!
+          touch_all
+          unscope
+          update_all
+          where
+          without
+        ].freeze
+
+        def on_send(node)
+          query_node = node.parent
+
+          return unless query_node&.send_type?
+          return unless QUERYING_METHODS.include?(query_node.method_name)
+          return if node.receiver.nil? && !inherit_active_record_base?(node)
+
+          range_of_all_method = node.loc.selector
```

Comment:
It should also be able to handle the following cases:
https://github.com/rubocop/rails-style-guide/pull/340#discussion_r1291407097

Reply:
I've already addressed cases where an Active Record query method follows `all`, as shown below:

```ruby
# bad
current_user.likes.all.order(:created_at)

# good
current_user.likes.order(:created_at)
```
https://github.com/rubocop/rubocop-rails/pull/1020/files#diff-8d2292cab6c23118ba7721e07148d49a27043e9202c75084bd49b25053a1575eR216

Should I also consider addressing cases where the receiver is a relation and no method follows `all`?

```ruby
# bad
current_user.likes.all

# good
current_user.likes
```

Reply:
Oops, this is my mistake. No problem!